### DISCUSSION TITLE
I hope there can be a lightweight version, just a storage bucket

### DISCUSSION THREAD

**Discussion by renjiali** [supabase/supabase#27366] [Feature Requests]
In many cases, we don't need too many things, such as databases. We just need its storage bucket.

**Comment by encima** [comment]
Hi there, thanks for opening.

Are the cases you are talking about for Supabase? Supabase itself a platform with each component pretty integrated into it. For example, Storage uses the database to store the object metadata and integrates with Auth to control access.

When you do not need to use some features, you are welcome to not use them (i.e. some people use a different Auth provider). In the self-hosted version, each component uses pretty minimal resources, but most are needed.

If you are talking about the repository size, we have a guide in our [docs](https://supabase.com/docs/guides/self-hosting/docker) under the `Advanced` tab that only checks out the docker folder.

In those cases where you just need storage, have you considered using [Minio](https://min.io/)?

**Reply by rjl-cli** [reply]
**I prepared two languages, Chinese and English, Chinese is the most original text. English is translated by AI**

### 中文
是，我曾经使用过，但是minio的一些api和逻辑不太有效。和supabase相比就太难用了。如果全量部署，低配置的服务器，直接卡死了，直接负载100%(使用的是2核2G 10M带宽) 我改成2核4G就是分正常
而且，我觉得这个要是可以支持私有存储桶。它的市场前景是非常好的。因为购买cdn 一是成本太大，二是又盗刷的风险。但是但多数人仅仅是要做为一个文件存储，方便对文件进行管理
### English
Yes, I have used it before, but some of Minio's APIs and logic are not very effective. Compared to Supabase, it is too difficult to use. If fully deployed on a low-configured server, it will be completely blocked and the load will reach 100% (using a 2-core 2G 10Mbps connection). Changing to a 2-core 4G configuration will make it normal.

Furthermore, I think that if it can support private storage buckets, its market prospect is very good. Because buying a CDN is not only expensive, but also has the risk of fraud. However, most people only want to use it as a file storage to manage files conveniently.

**Reply by rjl-cli** [reply]
Or whether it is simple to configure and deploy by masking some functions

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.

**Reply by rjl-cli** [reply]
**I prepared two languages, Chinese and English, Chinese is the most original text. English is translated by AI**

### **中文**
是，我曾经使用过，但是minio的一些api和逻辑不太有效。和supabase相比就太难用了。如果全量部署，低配置的服务器，直接卡死了，直接负载100%(使用的是2核2G 10M带宽) 我改成2核4G就是分正常
而且，我觉得这个要是可以支持私有存储桶。它的市场前景是非常好的。因为购买cdn 一是成本太大，二是又盗刷的风险。但是但多数人仅仅是要做为一个文件存储，方便对文件进行管理

### **English**
Yes, I have used it before, but some of Minio's APIs and logic are not very effective. Compared to Supabase, it is too difficult to use. If fully deployed on a low-configured server, it will be completely blocked and the load will reach 100% (using a 2-core 2G 10Mbps connection). Changing to a 2-core 4G configuration will make it normal.

Furthermore, I think that if it can support private storage buckets, its market prospect is very good. Because buying a CDN is not only expensive, but also has the risk of fraud. However, most people only want to use it as a file storage to manage files conveniently.

**Reply by rjl-cli** [reply]
Or whether it is simple to configure and deploy by masking some functions

**Comment by rjl-cli** [comment]
### I prepared two languages, Chinese and English, Chinese is the most original text. English is translated by AI

### 中文
是，我曾经使用过，但是minio的一些api和逻辑不太有效。和supabase相比就太难用了。如果全量部署，低配置的服务器，直接卡死了，直接负载100%(使用的是2核2G 10M带宽) 我改成2核4G就是分正常
而且，我觉得这个要是可以支持私有存储桶。它的市场前景是非常好的。因为购买cdn 一是成本太大，二是又盗刷的风险。但是但多数人仅仅是要做为一个文件存储，方便对文件进行管理

### English
Yes, I have used it before, but some of Minio's APIs and logic are not very effective. Compared to Supabase, it is too difficult to use. If fully deployed on a low-configured server, it will be completely blocked and the load will reach 100% (using a 2-core 2G 10Mbps connection). Changing to a 2-core 4G configuration will make it normal.

Furthermore, I think that if it can support private storage buckets, its market prospect is very good. Because buying a CDN is not only expensive, but also has the risk of fraud. However, most people only want to use it as a file storage to manage files conveniently.
